aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -118,6 +118,51 @@ SparseTensor& log1p_sparse_(SparseTensor& t) {
   return log1p_out_sparse(t, t);
 }
 
+// --------------------------------------------------------------------
+// neg(SparseTensor)
+// --------------------------------------------------------------------
+
+SparseTensor& neg_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  AT_ASSERT(r.is_sparse());
+  AT_ASSERT(t.is_sparse());
+
+  // copy_sparse_ does not perform the copy if it is the same tensor
+  copy_sparse_to_sparse_(r, t);
+  r._values().neg_();
+  return r;
+}
+
+SparseTensor& neg_sparse_(SparseTensor& t) {
+  return neg_out_sparse(t, t);
+}
+
+// --------------------------------------------------------------------
+// asin(SparseTensor)
+// --------------------------------------------------------------------
+
+// In-place asin on uncoalesced tensors is not supported due non-linear type of operation.
+// Values of uncoalesced tensor corresponding to the same indices are summed
+// and asin(summed_value) != asin(v1) + asin(v2)
+
+SparseTensor& asin_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  AT_ASSERT(r.is_sparse());
```

Comment:
AT_ASSERT -> TORCH_CHECK

Reply:
(AT_ASSERT is only appropriate if the assertion can only be triggered by internal errors, not user inputs. Also, AT_ASSERT is deprecated in favor of TORCH_INTERNAL_ASSERT.)